### PR TITLE
Replace NamedRunnable.java with a threadName function

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: mockwebserver/src/main/java/okhttp3/mockwebserver/MockWebServer.kt [hunk 1]
```diff
@@ -331,55 +331,53 @@ class MockWebServer : ExternalResource(), Closeable {
...
     port = serverSocket!!.localPort
-    executor!!.execute(object : NamedRunnable("MockWebServer $port") {
-      override fun execute() {
-        try {
-          logger.info("${this@MockWebServer} starting to accept connections")
-          acceptConnections()
-        } catch (e: Throwable) {
-          logger.log(Level.WARNING, "${this@MockWebServer}  failed unexpectedly", e)
-        }
+    executor!!.execute("MockWebServer $port") {
```
Comment (swankjesse) [inline mockwebserver/src/main/java/okhttp3/mockwebserver/MockWebServer.kt]: GitHub doesn’t do this diff well. It’s a 17-line diff when you ignore whitespace.

Comment (JakeWharton) [reply mockwebserver/src/main/java/okhttp3/mockwebserver/MockWebServer.kt]: https://github.com/square/okhttp/pull/5038/files?w=1

File: okhttp/src/main/java/okhttp3/internal/UtilKt.kt [hunk 2]
```diff
@@ -57,13 +57,23 @@ inline fun ignoreIoExceptions(block: () -> Unit) {
   }
 }
 
+inline fun threadName(name: String, block: () -> Unit) {
+  val oldName = Thread.currentThread().name
```
Comment (JakeWharton) [inline okhttp/src/main/java/okhttp3/internal/UtilKt.kt]: Is `Thread.currentThread()` expensive? Could make a local

File: okhttp/src/main/java/okhttp3/RealCall.kt [hunk 3]
```diff
@@ -115,7 +115,7 @@ internal class RealCall private constructor(
       assert(!Thread.holdsLock(client.dispatcher()))
       var success = false
       try {
-        executorService.execute(this)
+        executorService.execute("OkHttp ${redactedUrl()}", this@AsyncCall::run)
```
Comment (yschimke) [inline okhttp/src/main/java/okhttp3/RealCall.kt]: This is breaking the test expectations of RecordingExecutor :(

Comment (yschimke) [reply okhttp/src/main/java/okhttp3/RealCall.kt]: I was trying to be helpful but maybe overstepped a line on the PR.  No issue if you revert :)

File: okhttp/src/main/java/okhttp3/internal/UtilKt.kt (hunk 2 above)
Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/internal/UtilKt.kt]: Fixed.

File: okhttp/src/main/java/okhttp3/RealCall.kt (hunk 3 above)
Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/RealCall.kt]: yeah, I fixed

**Pull request merged**